components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -212,18 +257,36 @@ def run_clp(
 
     # Compute the total amount of data compressed
     last_archive_stats = None
-    total_uncompressed_size = 0
-    total_compressed_size = 0
-    while True:
+    last_line_decoded = False
+    worker_output = {
+        "total_uncompressed_size": 0,
+        "total_compressed_size": 0,
+    }
+
+    while not last_line_decoded:
         line = proc.stdout.readline()
-        if not line:
-            break
-        stats = json.loads(line.decode("ascii"))
-        if last_archive_stats is not None and stats["id"] != last_archive_stats["id"]:
+        stats: Optional[Dict[str, Any]] = None
+        if line:
+            stats = json.loads(line.decode("ascii"))
+        else:
+            last_line_decoded = True
+
+        if last_archive_stats is not None and (
+            None is stats or stats["id"] != last_archive_stats["id"]
+        ):
+            if enable_s3_write:
+                result = upload_single_file_archive_to_s3(
+                    last_archive_stats, archive_output_dir, s3_config
+                )
+                if not result.success:
+                    worker_output["error_message"] = result.error
+                    s3_write_failed = True
+                    break
+
             # We've started a new archive so add the previous archive's last
             # reported size to the total
-            total_uncompressed_size += last_archive_stats["uncompressed_size"]
-            total_compressed_size += last_archive_stats["size"]
+            worker_output["total_uncompressed_size"] += last_archive_stats["uncompressed_size"]
+            worker_output["total_compressed_size"] += last_archive_stats["size"]
```

Comment:
Can we go back to using variables for these so that there's no risk of typos in the keys (except in the single place we set them)?